### PR TITLE
[MRG] Add new acquisition function that chooses the start point of lbfgs wisely

### PR DESCRIPTION
**Pull request by MechCoder**

### CODE REVIEW

**Comment by MechCoder** [comment]
New benchmarks:

On branin: 50 calls and averaged across 10 runs


| Acquisition Function | Mean Minimum | Std |
| --- | --- | --- |
| this branch | 0.3992 |  0.0011 |
| lbfgs | 0.4104 | 0.0146 |
| sampling | 0.3995   | 0.0014 |

**Comment by MechCoder** [comment]
On hart6:

| Acquisition Function | Mean Minimum | Std |
| --- | --- | --- |
| this branch | -3.2273 |  0.0617 |
| lbfgs | -2.9502 | 0.3273 |
| sampling | -3.1416   | 0.0646 |

File: skopt/optimizer/base.py [hunk 1]
```diff
@@ -277,18 +278,25 @@ def base_minimize(func, dimensions, base_estimator,
 
         models.append(gp)
 
-        if acq_optimizer == "sampling":
+        if acq_optimizer == "sampling" or acq_optimizer == "lbfgs_optim":
```
Comment (MechCoder) [inline skopt/optimizer/base.py]: It might be better to implement this in "lbfgs" itself rather than create a separate acquisition function.

Comment (glouppe) [inline skopt/optimizer/base.py]: +1, let's simply things by having only two options for `acq_optimizer`.

**Comment by glouppe** [comment]
Shall we harmonize `n_points` and `n_restarts_optimizer` to have the same default values for all our optimizers?

**Comment by glouppe** [comment]
Other than that, results look good in my opinion! This would be a nice compromise for closing #262. What do you think? (the added-value of spray points being left out for further study)

**Comment by MechCoder** [comment]
Good to go from my side. Let us begin the new year on a good note :P

**Comment by glouppe** [comment]
Travis is happy! Merging :)

**Pull request merged**